src/clr-core/modal/modal.element.scss

```diff
@@ -0,0 +1,120 @@
+@import './../styles/tokens/generated/index';
+@import './../styles/mixins/utils';
+@import './../styles/utils/close';
+@import './../styles/utils/a11y';
+
+:host {
+  --backdrop-opacity: 0.85;
+  --backdrop-color: #{$cds-token-color-neutral-900};
+  --box-shadow-color: #{$cds-token-color-action-500};
+  --border-radius: #{$cds-token-global-border-radius};
+  --border-color: #{$cds-token-color-action-600};
+  --background-color: #{$cds-token-color-neutral-0};
+  --close-icon-color: #{$cds-token-color-neutral-600};
+  --close-icon-color-hover: #{$cds-token-color-neutral-700};
+  --content-box-shadow-color: #{$cds-token-color-neutral-1000};
+  --title-color: #{$cds-token-typography-h3-color};
+  --title-font-family: #{$cds-token-typography-header-font-family};
+  --title-font-weight: #{$cds-token-typography-h3-font-weight};
+  --width: calc(8 * #{$cds-token-space-size-13});
+}
+
+:host([size='sm']) {
+  --width: calc(4 * #{$cds-token-space-size-13});
+}
+
+:host([size='lg']) {
+  --width: calc(12 * #{$cds-token-space-size-13});
+}
+
+:host([size='xl']) {
+  --width: calc(16 * #{$cds-token-space-size-13});
+}
+
+.private-host {
+  position: fixed;
+  top: 0;
+  bottom: 0;
+  right: 0;
+  left: 0;
+  z-index: 1050; // TODO: restore using map-get to get the z-index
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+}
+
+button.close {
+  font-size: $cds-token-space-size-8;
+  line-height: $cds-token-space-size-8;
+
+  &:focus,
+  &:hover,
+  &:active {
+    --close-icon-color: var(--close-icon-color-hover);
+  }
+
+  cds-icon {
+    fill: var(--close-icon-color);
+    // per measurement, this results in an icon that is 16x16...
+    @include equilateral($cds-token-space-size-8);
+  }
+}
+
+.modal-dialog {
+  position: relative;
+  z-index: 1050; // TODO: restore using map-get to get the z-index
+  width: var(--width);
+  max-width: 100%;
+
+  $modal-boxshadow-y: $cds-token-space-size-1;
+  $modal-boxshadow-spread: $cds-token-space-size-2;
+
+  .modal-content {
```

Comment:
the `.modal-content` selector does not need to be nested inside the `.modal-dialog` selector.

Why do we need the classnames?

Reply:
Moved it out! 

I don't know if we need to keep these sass variables (carried over from Angular). I can get rid of them and use the tokens directly. Or are you asking why `.modal-dialog` and `.modal-content` classes exist?

Reply:
If we have `cds-modal-content` and other components, why do we need to use the `.modal-content` classname?

I don't see internal wrappers in those components (like i've used to help with layouts and LHEs in tags and buttons) so I don't understand why we can't use the component tagName as the selector.

Reply:
since these classnames are not direct parent of each selector, I think they are still needed in these specific cases. Take a look and let me know if you think otherwise.

Reply:
We don't have to carryover the SASS variables.

Reply:
Removing the variables!